src/libraries/Microsoft.PowerFx.Core/Parser/TokenCursor.cs

```diff
@@ -103,11 +121,26 @@ public Token TokMove()
         {
             AssertValid();
             var tok = _currentToken;
+            
             if (_currentTokenId != TokKind.Eof)
             {
                 MoveTo(_currentTokenIndex + 1);
             }
 
+            var tokenLength = tok.Span.Lim - tok.Span.Min;
+            _currentCharIndex += tokenLength;
+
+            if (_shouldUseOffset)
+            {
+                tok = tok.Clone(new Localization.Span(_offset, _offset + tokenLength));
+                if (tok.Kind == TokKind.Semicolon)
+                {
+                    _shouldUseOffset = false;
+                }
```

Comment:
We shouldn't do this check.  Semi could appear within a formula, and not necessarily indicate the end of a formula.